fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/postgresql/PostgreSqlResourceDAO.java

```diff
@@ -0,0 +1,533 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.jdbc.postgresql;
+
+import static com.ibm.fhir.persistence.jdbc.JDBCConstants.UTC;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLIntegrityConstraintViolationException;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.UUID;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.transaction.TransactionSynchronizationRegistry;
+
+import com.ibm.fhir.database.utils.postgresql.PostgreSqlTranslator;
+import com.ibm.fhir.persistence.exception.FHIRPersistenceException;
+import com.ibm.fhir.persistence.exception.FHIRPersistenceVersionIdMismatchException;
+import com.ibm.fhir.persistence.jdbc.dao.api.CodeSystemDAO;
+import com.ibm.fhir.persistence.jdbc.dao.api.FhirRefSequenceDAO;
+import com.ibm.fhir.persistence.jdbc.dao.api.ParameterDAO;
+import com.ibm.fhir.persistence.jdbc.dao.api.ParameterNameDAO;
+import com.ibm.fhir.persistence.jdbc.dao.impl.ParameterVisitorBatchDAO;
+import com.ibm.fhir.persistence.jdbc.dao.impl.ResourceDAOImpl;
+import com.ibm.fhir.persistence.jdbc.derby.CodeSystemCacheAdapter;
+import com.ibm.fhir.persistence.jdbc.derby.DerbyCodeSystemDAO;
+import com.ibm.fhir.persistence.jdbc.derby.DerbyParameterNamesDAO;
+import com.ibm.fhir.persistence.jdbc.derby.FhirRefSequenceDAOImpl;
+import com.ibm.fhir.persistence.jdbc.derby.ParameterNameCacheAdapter;
+import com.ibm.fhir.persistence.jdbc.dto.ExtractedParameterValue;
+import com.ibm.fhir.persistence.jdbc.dto.Resource;
+import com.ibm.fhir.persistence.jdbc.exception.FHIRPersistenceDBConnectException;
+import com.ibm.fhir.persistence.jdbc.exception.FHIRPersistenceDataAccessException;
+import com.ibm.fhir.persistence.jdbc.exception.FHIRPersistenceFKVException;
+import com.ibm.fhir.persistence.jdbc.util.ResourceTypesCache;
+
+/**
+ * Data access object for writing FHIR resources to an Apache PostgreSql database.
+ *
+ * @implNote The original implementation (for DSTU2) used a global temporary table
+ * to pass the parameter list into the stored procedure, but this approach
```

Comment:
Do we want to use DSTU2 reference? I think we should remove.

Reply:
Also we probably just want to reference db2, we haven't used GTTs recently.

Reply:
updated the comments, done